url-updater/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -431,12 +442,30 @@ private void doUpdateStatusJson(boolean success, int statusCode, UrlVersion urlV
     }
   }
 
+  /**
+   * Checks if the timeout was expired.
+   *
+   * @return boolean true if timeout was expired, false if not
+   */
+  public boolean isTimeoutExpired() {
+
+    if (this.expirationTime == null) {
+      return false;
+    }
+
+    if (Instant.now().isAfter(expirationTime)) {
+      logger.warn("Expiration time of timeout was reached, cancelling update process.");
+      return true;
+    }
+
+    return false;
+  }
+
   /**
    * Updates the tool's versions in the URL repository.
    *
    * @param urlRepository the {@link UrlRepository} to update
    */
-  @Override
```

Comment:
why did you remove this override? The method is still overridden. The annotation is used to ensure this.

Reply:
Was by mistake. Re-added it now.